### ISSUE TITLE
Spurious warnings for methods in Objective-C categories

### ISSUE THREAD

**Issue by jlstrecker** [doxygen/doxygen#10101]
**Describe the bug**
Doxygen emits warnings ("A define (_method_) cannot be made a member of _class_") for methods in Objective-C categories.

**Expected behavior**
A warning should not be reported here, since the code and documentation are valid.

**To Reproduce**
Unzip [objc_method_test.zip](https://github.com/doxygen/doxygen/files/11593962/objc_method_test.zip) and run `doxygen`.
```
objc_method_test/bad.h:0: warning: A define (doStuff) cannot be made a member of TestClass
```

**Version**
Doxygen 1.9.7, macOS, Homebrew

The bug did not happen in Doxygen 1.9.6 or earlier.

**Comment by albert-github** [comment]
Looks like a regression, a bisect lead to:
```
535e5b8d70ed3735e8f885cc9bca10da7bb4b2d1 is the first bad commit
commit 535e5b8d70ed3735e8f885cc9bca10da7bb4b2d1

Date:   Thu Apr 20 21:34:35 2023 +0200

    Refactoring: remove unneeded m_impl indirection from MemberDefImpl

 src/memberdef.cpp | 1254 ++++++++++++++++++++++++++---------------------------
 1 file changed, 612 insertions(+), 642 deletions(-)
```

**Comment by doxygen** [comment]
@jlstrecker Please verify that the referenced commit fixed the problem for you. Do not close the issue, this will be done automatically when the next official version is released.

**Comment by jlstrecker** [comment]
Confirmed fixed. Thanks.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
